Add new information to Validium page (#6502)
ethereum/ethereum-org-website: src/content/developers/docs/scaling/validium/index.md

sidebarDepth: 3
---

> Validium is a [scaling solution](/developers/docs/scaling/) that enforces integrity of transactions using validity proofs like [ZK-rollups](/developers/docs/scaling/zk-rollups/), but doesn’t store transaction data on the main Ethereum layer. While off-chain data availability introduces trade-offs, it can lead to massive improvements in scalability (validiums can process ~ 9,000 transactions, or more, per second). 

Reviewer: > doesn’t store transaction data on the main Ethereum layer

This reads a bit vague to me - should we be explicit to say "doesn't store transaction data on chain"?

Author: I felt saying validiums don't store data "on-chain" at the beginning of the page would throw off readers ("what chain are we referring to?"). Establishing the main point (i.e., validiums don't store data on Ethereum) from the start (as opposed to ZK-rollups) was the priority here. Also, I think the idea of on-chain vs off-chain storage is well discussed in the other sections, so readers are likely to understand the concept.

Reviewer 2: I think we can be fairly explicit and accurate here imo. I would argue the page we're editing is in the "Advanced" section of the docs, and if we're going to refer to where the data is being stored, I would rather we use proper terms like "Mainnet execution layer" or just "Ethereum Mainnet" over "main Ethereum layer" for example.
I think using "on-chain" is fine, but to your point if you want to clarify we're referring to Ethereum Mainnet here I think that's perfectly reasonable.

Reviewer 2: ```suggestion
Validium is a [scaling solution](/developers/docs/scaling/) that enforces integrity of transactions using validity proofs like [ZK-rollups](/developers/docs/scaling/zk-rollups/), but doesn’t store transaction data on the Ethereum Mainnet. While off-chain data availability introduces trade-offs, it can lead to massive improvements in scalability (validiums can process ~ 9,000 transactions, or more, per second). 
```

## Prerequisites {#prerequisites}

You should have a good understanding of all the foundational topics and a high-level understanding of [Ethereum scaling](/developers/docs/scaling/). Implementing scaling solutions such as Validium is an advanced topic as the technology is less battle-tested, and continues to be researched and developed.

## How does validium work? {#how-does-validium work}

> Validiums are a type of “rollup”: a [Layer 2](/layer-2/) scaling protocol designed to improve throughput by processing transactions off the main Ethereum chain. Off-chain transactions executed on the validium chain are verified via a smart contract on the base Ethereum layer using zero-knowledge proofs. This can be either a SNARK (Succinct Non-Interactive Argument of Knowledge) or SNARK (Scalable Transparent Argument of Knowledge). 

Reviewer: ```suggestion
Validiums are a type of “rollup”: a [Layer 2](/layer-2/) scaling protocol designed to improve throughput by processing transactions off the main Ethereum chain. Off-chain transactions executed on the validium chain are verified via a smart contract on the base Ethereum layer using zero-knowledge proofs. This can be either a SNARK (Succinct Non-Interactive Argument of Knowledge) or STARK (Scalable Transparent ARgument of Knowledge). 
```

Reviewer: Might be nice to link out to learning resources for SNARK & STARK.

Reviewer: e.g. https://consensys.net/blog/blockchain-explained/zero-knowledge-proofs-starks-vs-snarks/

sidebarDepth: 3
---

> Validium is a [scaling solution](/developers/docs/scaling/) that enforces integrity of transactions using validity proofs like [ZK-rollups](/developers/docs/scaling/zk-rollups/), but doesn’t store transaction data on the Ethereum Mainnet. While off-chain data availability introduces trade-offs, it can lead to massive improvements in scalability (validiums can process ~ 9,000 transactions, or more, per second). 

Reviewer 3: Works for me, just wanted to make sure we had something to back the claim if we are going to have that.

## Prerequisites {#prerequisites}

You should have a good understanding of all the foundational topics and a high-level understanding of [Ethereum scaling](/developers/docs/scaling/). Implementing scaling solutions such as Validium is an advanced topic as the technology is less battle-tested, and continues to be researched and developed.

## How does validium work? {#how-does-validium work}

> Validiums are a type of “rollup”: a [Layer 2](/layer-2/) scaling protocol designed to improve throughput by processing transactions off the main Ethereum chain. Off-chain transactions executed on the validium chain are verified via a smart contract on the base Ethereum layer using zero-knowledge proofs, which can be SNARKs (Succinct Non-Interactive Argument of Knowledge) or STARKs (Scalable Transparent ARgument of Knowledge). 

Reviewer 4: On ethereum.org, we don't consider a validium chain to be a layer 2 because it doesn't have on-chain data availability.

[From our listing policy:](https://ethereum.org/en/contributing/adding-layer-2/#criteria-for-inclusion-the-must-haves)

> _We do not consider other scaling solutions that don't use Ethereum for data availability or security to be layer 2._

Reviewer 2: Yeah good catch, agree we shouldn't label this as such.

More on [zero-knowledge proofs](https://consensys.net/blog/blockchain-explained/zero-knowledge-proofs-starks-vs-snarks/).

> As with ZK-rollups, validity proofs prevent invalid state transitions on validium chains and enhance security guarantees available to users. For instance, a malicious operator cannot steal funds since users can always withdraw funds directly from the on-chain contract [using Merkle proofs](/developers/tutorials/merkle-proofs-for-offline-data-integrity/). 

Reviewer 3: I think funds can be frozen if data availability isn't there. While a malicious operator cannot steal them, I think its important to still note that funds can be frozen as this is a big difference between zk-rollups and validiums.

The major difference between validiums and ZK-rollups concerns their positions on the data availability spectrum. Both solutions approach data storage differently, which has implications for security and trustlessness. 

> ### Off-chain vs on-chain data availability {#off-chain-vs-on-chain-data-availability}

Reviewer 3: ```suggestion
### On-chain vs off-chain data availability {#on-chain-vs-off-chain-data-availability}
```
Switching order since we open up with zk-rollups here

Reviewer 3: I think in this section we could categorize different off-chain data availability solutions similar to how we do use cases on the bridges page (https://ethereum.org/en/bridges/#bridge-use-cases)

Something like:
```
### on-chain vs off-chain data availability
zk-rollup onchain blurb
validium offchain blurb

#### Data availability committees (DAC)
DAC blurb with some resources

#### Bond data availability
blurb on bond data availability with some resources

Off-chain data availability does present a problem: data necessary for creating Merkle proofs may be unavailable. This means users may be unable to withdraw funds from the rollup contract if operators should act maliciously. 
```

> **ZK-rollups** post information about off-transactions to Ethereum Mainnet as `calldata`. This information can be used to recreate the state of the rollup, if necessary, to prove the validity (or lack thereof) of a specific state transition. As the data is anchored on Ethereum (**on-chain data availability**), ZK-rollups can directly benefit from Ethereum’s security guarantees—funds are always secure if Mainnet is operational.

Reviewer 3: ```suggestion
**ZK-rollups** bundle (or "roll-up") off-chain transaction data to Ethereum Mainnet as `calldata`. This information can be used to recreate the state of the rollup, if necessary, to prove the validity (or lack thereof) of a specific state transition. As the data is anchored on Ethereum (**on-chain data availability**), ZK-rollups directly benefit from Ethereum’s security and data availability guarantees—funds are always secure if Mainnet is operational.
```
Just cleaning up the first sentance, and adding a bit about data availability as well.

> ZK-rollups post information about off-transactions to Ethereum Mainnet as `calldata`. This information can be used to recreate the state of the rollup, if necessary, to prove the validity (or lack thereof) of a specific state transition. As the data is anchored on Ethereum (on-chain data availability), ZK-rollups can directly benefit from Ethereum’s security guarantees—funds are always secure if Mainnet is operational.
> 
> Validiums keep all data relating to transactions off the Ethereum chain (off-chain data availability). This improves throughput on validium chains since less data needs to be posted to Mainnet. It also reduces rollup costs by eliminating the need to publish calldata and increases privacy for users (transaction information doesn't make it to Mainnet). 

Reviewer 2: Styling suggestion, just to help people quickly visually realize these are comparative paragraphs and highlight the tl;dr

```suggestion
**ZK-rollups** post information about off-transactions to Ethereum Mainnet as `calldata`. This information can be used to recreate the state of the rollup, if necessary, to prove the validity (or lack thereof) of a specific state transition. As the data is anchored on Ethereum (**on-chain data availability**), ZK-rollups can directly benefit from Ethereum’s security guarantees—funds are always secure if Mainnet is operational.

**Validiums** keep all data relating to transactions off the Ethereum chain (**off-chain data availability**). This improves throughput on validium chains since less data needs to be posted to Mainnet. It also reduces rollup costs by eliminating the need to publish calldata and increases privacy for users (transaction information doesn't make it to Mainnet). 
```

> **Validiums** keep all data relating to transactions off the Ethereum chain (**off-chain data availability**). This improves throughput on validium chains since less data needs to be posted to Mainnet. It also reduces rollup costs by eliminating the need to publish calldata and increases privacy for users (transaction information doesn't make it to Mainnet). 

Reviewer 3: ```suggestion
**Validiums** store all transaction data off Ethereum Mainnet (**off-chain data availability**). This improves throughput and reduces the costs on validium chains since calldata is not posted to Ethereum Mainnet. 
```
Couple changes cleaning up sentences. I don't know that I agree with the increases privacy section, do you have a source you can link out to for this?

> Off-chain data availability does present a problem: data necessary for creating Merkle proofs may be unavailable. This means users may be unable to withdraw funds from the rollup contract if operators should act maliciously. 

Reviewer 3: ```suggestion
Off-chain data availability does present a problem: data necessary for creating or verifying Merkle proofs may be unavailable. This means users may be unable to withdraw funds from the rollup contract if operators should act maliciously. 
```

> To guarantee the availability of off-chain data, some validium solutions use a group of trusted entities to store copies of the state and provide proof of data availability. Of course, this presents another problem: users must trust these entities to make the data available when needed. There's also the possibility of members of data availability committees [getting compromised by a malicious actor](https://notes.ethereum.org/DD7GyItYQ02d0ax_X-UbWg?view) who can then withhold off-chain data. 

Reviewer 3: ```suggestion
To guarantee the availability of off-chain data, some validium solutions use a group of trusted entities know as a data availability committee (DAC) to store copies of the state and provide proof of data availability. Users must trust these entities to make the data available when needed. There's the possibility of members of data availability committees [getting compromised by a malicious actor](https://notes.ethereum.org/DD7GyItYQ02d0ax_X-UbWg?view) who can then withhold off-chain data. 
```

## Pros and cons of validium {#pros-and-cons-of-validium}
| Pros                                                                                                      | Cons                                                                                                                                     |
| --------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
| No withdrawal delay (no latency to on-chain/cross-chain tx); consequent greater capital efficiency.       | Limited support for general computation/smart contracts; specialized languages required.                                                 |
| Not vulnerable to certain economic attacks faced by fraud-proof based systems in high-value applications. | High computational power required to generate ZK proofs; not cost effective for low throughput applications.                             |

> | Reduces rollup fees for users by shrinking costs of publishing data on Ethereum.                          | Slower subjective finality time (10-30 min to generate a ZK proof) (but faster to full finality because there is no dispute time delay). |

Reviewer 3: ```suggestion
| Reduces gas fees for users by not posting calldata to Ethereum Mainnet.                          | Slower subjective finality time (10-30 min to generate a ZK proof) but faster to full finality because there is no dispute time delay. |
```

> | Suitable for specific use-cases, like trading or blockchain gaming that prioritize transaction privacy and scalability. | Generating a proof requires off-chain data to be available at all times.                                                   |

Reviewer 3: Does validium prioritize privacy?

Author: Sort of. As I mentioned earlier, StarkWare suggests validium is relatively more private for users than zk-rollups with on-chain data availability.

Author: > Does validium prioritize privacy?

We could change that point to read: "Suitable for specific use-cases...that require lower trust assumptions/trustlessness." I think that highlights the major difference between zk-rollups and validiums. LMK if that works.

Reviewer 2: I can see how Validiums could improve privacy from the same perspective as your transactions on a centralized exchange are "private." If the data isn't published it would be more private to the general public, but also increases the trust assumptions if you aren't the one holding that data yourself.
